### PR TITLE
Prefix: fixing the positive case

### PR DESCRIPTION
**Pull request by vhavlena**

### CODE REVIEW

**Comment by vhavlena** [comment]
Results:
```
# of formulae: 103405
--------------------------------------------------
tool                            ✅    ❌       time    avg    med    std    sat    unsat    unknown    TO    MO+ERR    other
--------------------------  ------  ----  ---------  -----  -----  -----  -----  -------  ---------  ----  --------  -------
z3-noodler-0122b33-17ffaf6  102443   962   12100.47   0.12   0.02   1.66  62952    39491        528   372        62        0
z3-noodler-5249443-17ffaf6  102431   974   12179.01   0.12   0.02   1.67  62942    39489        530   380        64        0
cvc5-1.2.0                   99841  3564   73861.19   0.74   0.01   6.42  60931    38910          2  3553         9        0
z3-4.13.0                    97745  5660  128712.47   1.32   0.01   7.95  58927    38818        211  4914       465       70
--------------------------------------------------
```

image

**Comment by jurajsic** [comment]
Nice, but you should  compare with `z3-noodler-193f136-17ffaf6` not with `z3-noodler-5249443-17ffaf6` (I forgot to push it).

However, it seems like there is still a bug somewhere else, because on the original formula `QF_SLIA/2019-full_str_int/py-conbyte_z3seq/leetcode_int-validWordAbbreviation/162.smt2` it still gives incorrect model (probably also on `QF_SLIA/2019-full_str_int/py-conbyte_z3seq/leetcode_int-validWordAbbreviation/36.smt2`, I did not check).

File: src/smt/theory_str_noodler/theory_str_noodler.cpp [hunk 1]
```diff
@@ -191,14 +191,17 @@ namespace smt::noodler {
         }
         // we need to propagate not(prefix) and not(suffix) before the actual solve (in init_search), because we need to ensure these axioms are
```
Comment (jurajsic) [inline src/smt/theory_str_noodler/theory_str_noodler.cpp:192]: I would update the comment, it only talks about the negated versions and only about suffix/prefix.

**Comment by vhavlena** [comment]
Sure, updated results
```
# of formulae: 103405
--------------------------------------------------
tool                            ✅    ❌       time    avg    med    std    sat    unsat    unknown    TO    MO+ERR    other
--------------------------  ------  ----  ---------  -----  -----  -----  -----  -------  ---------  ----  --------  -------
z3-noodler-0122b33-17ffaf6  102443   962   12100.47   0.12   0.02   1.66  62952    39491        528   372        62        0
z3-noodler-193f136-17ffaf6  102442   963   12345.08   0.12   0.02   1.72  62953    39489        530   372        61        0
cvc5-1.2.0                   99841  3564   73861.19   0.74   0.01   6.42  60931    38910          2  3553         9        0
z3-4.13.0                    97745  5660  128712.47   1.32   0.01   7.95  58927    38818        211  4914       465       70
--------------------------------------------------
```

image

File: src/smt/theory_str_noodler/theory_str_noodler.cpp (hunk 1 above)
Comment (vhavlena) [reply src/smt/theory_str_noodler/theory_str_noodler.cpp]: done

**Review by jurajsic — APPROVED** [review]

**Pull request merged**